lib/model.js

```diff
@@ -347,6 +347,51 @@ var Model = define([QueryPlugin, Middleware], {
         },
 
         /**
+         * Convert this model to an object including any associations passed in.
+         *
+         * This will not handle recursive associations and is not useful when working with
+         * eager loaded models.
+         *
+         * @return {Object} the object version of this model.
+         **/
+        toObjectWithAssociations: function(associations) {
+            if (!associations) {
+                return this.toObject();
+            }
+
+            var includes = [],
+                modelInstance = this,
+                json = this.toObject(),
+                associationJson = {};
+
+            if (Array.isArray(associations)) {
+                includes = includes.concat(associations);
+            } else {
+                includes.push(associations);
+            }
+
+            var whens = (includes || []).map(function(associationName) {
+                return modelInstance[associationName].chain(function(association) {
```

Comment:
Do we need a `null` check for `modelInstance[associationName]` before we call `chain`?